brave/brave-core
# Simplifies userdrain.

Resolves https://github.com/brave/brave-browser/issues/20554.

This PR removes the `promotion` table dependency from the userdrain process.
All promotion tokens (that is, `ADS` and `UGP`) are now submitted to `/v2/suggestions/claim`.
`UGP` tokens are taken in and marked as errored on the server side, client receives `HTTP 200 (OK)`.

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. clean profile
2. opt-in to Brave Rewards on `brave://rewards`
3. claim the `UGP` grants on staging (`30 BAT`)
4. have someone send you `ADS` grants on staging (`x BAT`) and claim those, too (can also contact me — @szilardszaloki)
5. turn off the browser
6. remove all records from the `promotion` table (execute `DELETE FROM promotion` in `publisher_info_db`)
7. turn on the browser
8. try verifying your Rewards wallet with a wallet provider

Make sure:
- `UGP` grants (`30 BAT`) don't show up in `Rewards BAT` under __Balance info__ on `brave://rewards-internals`
- `UGP` grants (`30 BAT`) don't show up in the balance of your chosen wallet provider under __Balance info__ on `brave://rewards-internals`
- `ADS` grants (`x BAT`) do show up in the balance of your chosen wallet provider under __Balance info__ on `brave://rewards-internals`
